src/hooks/hooks.ts

```diff
@@ -0,0 +1,8 @@
+import { useDispatch, useSelector } from "react-redux";
+import type { TypedUseSelectorHook } from "react-redux";
+import type { RootState, AppDispatch } from "../store/store";
+
+export const useAppDispatch: () => AppDispatch = useDispatch;
+
+export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
```

Comment:
This is useless too, types will be inferred from the usage based on the passed selector. Was it your idea or you saw it somewhere? Am I missing something?

Reply:
https://redux-toolkit.js.org/usage/usage-with-typescript#getting-the-dispatch-type